src/rt/sections_elf_shared.d

```diff
@@ -906,8 +907,27 @@ version (Shared) void* handleForAddr(void* addr) nothrow @nogc
  */
 struct tls_index
 {
-    size_t ti_module;
-    size_t ti_offset;
+    version (CRuntime_Glibc)
+    {
+        // For x86_64, fields are of type uint64_t, this is important for x32
+        // where tls_index would otherwise have the wrong size.
+        // See https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/x86_64/dl-tls.h
+        version (X86_64)
+        {
+            ulong ti_module;
+            ulong ti_offset;
+        }
+        else
+        {
+            c_ulong ti_module;
+            c_ulong ti_offset;
```

Comment:
I've verified that in other cases `ti_module` & `ti_offset` are `unsigned long int`.

Reply:
Oh, just for the discussion thread, I left `size_t` in as the default branch.  At least musl defines this as a `size_t[2]` type.

(Grep'ing musl 1.16 for __tls__get_addr):
```
void *__tls_get_addr(size_t *);
...
return __tls_get_addr((size_t []){def.dso->tls_id, def.sym->st_value});
```

Reply:
By "other cases" I meant only by grepping the source of glibc. I didn't mean for musl etc.